### DISCUSSION TITLE
Lesson 6: IndexError @ 5:42:02 timestamp

### DISCUSSION THREAD

**Discussion by ArcusConsulting** [smartcontractkit/full-blockchain-solidity-course-py#345] [Q&A] [answered]
When I run brownie run scripts/deploy.py --networks ganache-local I get the following:

![image](https://user-images.githubusercontent.com/91853555/141382356-edae973e-8c82-4b13-8d47-48fb9b3bd936.png)

Here is my deploy.py:

![image](https://user-images.githubusercontent.com/91853555/141382408-ac8a9135-0714-4a9f-8ad2-67ed4fb4af48.png)

MY helpful_scripts.py:

![image](https://user-images.githubusercontent.com/91853555/141382462-06e4bb01-fbe3-4251-9833-597d24eb9f36.png)

My brownie-config.yaml:
![image](https://user-images.githubusercontent.com/91853555/141382521-24d90a5a-a076-41d5-b53b-3301cf017f76.png)

... and the command to add the ganache-local network to the network list:

![image](https://user-images.githubusercontent.com/91853555/141382635-f5d40375-99f0-4907-9547-313f9f9e5484.png)

**Comment by cromewar** [comment, marked as answer]
Hello @ArcusConsulting  you had added the ganache-local to your brownie networks but you didn't add the number of accounts, so brownie is complaining with index out of range because there aren't any accounts. So use:

`brownie networks add Ethereum ganache-local host=yourhost chainid=theid accounts=10`.

Hope this info might be helpful, best regards.

Note: remember to delete the network with `brownie networks delete ganache-local` before adding it again.

**Reply by ArcusConsulting** [reply]
I get an error:

ValueError: Unknown field(s): accounts

**Reply by ArcusConsulting** [reply]
Turns out I had a typo in my host address! Resolved and everything is working. Thanks!

**Reply by cromewar** [reply]
I'm glad that helped, I'm going to set this thread as solved. Feel free to post any other questions you might have.

**Comment by Vonnie187** [comment]
unfortunately this doesn't solve the same issue for me...
![helpful_scripts](https://user-images.githubusercontent.com/16199365/177030809-0cc4f67b-f687-45a6-bff3-51297fd422bd.png)
![add_local_ganache_accounts error](https://user-images.githubusercontent.com/16199365/177030813-fa65a1aa-dd3c-4324-bb50-2389e5d97e60.png)
![add_local_ganache](https://user-images.githubusercontent.com/16199365/177030814-55c712bf-8462-4d5b-92a4-d34b6418a7f0.png)

**Reply by cromewar** [reply]
is strange, but the default number of accounts is actually 10, so you might not worry about it.

**Reply by Vonnie187** [reply]
I've tried it all ways but i'm too new to understand where the index error is coming from or why its out of range.
We state account[0] and the default is account 0; so no idea but its cramping my progression because i need to know why lol

**Reply by cromewar** [reply]
just a question, your ganache local refer to the UI instance of ganache?

**Reply by fulltimecrypto** [reply]
Seems like I have the same issue as @Vonnie187, unfortunately...

**Reply by Vonnie187** [reply]
> just a question, your ganache local refer to the UI instance of ganache?

I'm not sure how to check, it could be possible that ive installed ganache UI outside the scope if my WSL? I just don't know eneough yet to find out why, driving me crazy lol

**Reply by cromewar** [reply]
@Vonnie187 
As you are using WSL you just need the change the host in Ganache UI to WSL one, you will find that on the setting tab.

**Reply by Cosmodude** [reply]
> @Vonnie187 As you are using WSL you just need the change the host in Ganache UI to WSL one, you will find that on the setting tab.

Have the same problem.
Changed the host to "All interfaces", still doesn't work

**Reply by Cosmodude** [reply]
Just solved the problem by changing the ganache host from 0.0.0.0 (All interfaces) to 127.0.0.1 (loopback pseudo interface), deleting and  reconnecting the network
